utilities/pipelines/resourceRemoval/helper/Invoke-ResourceRemoval.ps1

```diff
@@ -27,62 +85,78 @@ function Invoke-ResourceRemoval {
         [string] $Type
     )
 
-    switch ($type) {
+    Remove-UnhandledDeleteLock -ResourceId $ResourceId
+
+    switch ($Type) {
         'Microsoft.Insights/diagnosticSettings' {
-            $parentResourceId = $resourceId.Split('/providers/{0}' -f $type)[0]
+            $parentResourceId = $ResourceId.Split('/providers/{0}' -f $Type)[0]
             $resourceName = Split-Path $ResourceId -Leaf
             if ($PSCmdlet.ShouldProcess("Diagnostic setting [$resourceName]", 'Remove')) {
                 $null = Remove-AzDiagnosticSetting -ResourceId $parentResourceId -Name $resourceName
             }
             break
         }
         'Microsoft.Authorization/locks' {
-            $lockName = ($resourceId -split '/')[-1]
-            $lockScope = ($resourceId -split '/providers/Microsoft.Authorization/locks')[0]
+            $lockName = ($ResourceId -split '/')[-1]
+            $lockScope = ($ResourceId -split '/providers/Microsoft.Authorization/locks')[0]
 
             $null = Remove-AzResourceLock -LockName $lockName -Scope $lockScope -Force
-            Write-Verbose "Removed lock [$resourceName]. Waiting 10 seconds for propagation." -Verbose
-            Start-Sleep 10
+
+            $retryCount = 0
+            $retryLimit = 10
+            $retryInterval = 10
+            do {
+                $retryCount++
+                if ($retryCount -ge $retryLimit) {
+                    Write-Warning ('    [!] Lock [{0}] was not removed after {1} seconds. Continuing with resource removal.' -f $lockName, ($retryCount * $retryInterval))
+                    break
+                }
+                Write-Verbose ('    [⏱️] Waiting for lock [{0}] to be removed.' -f $lockName) -Verbose
+                Start-Sleep -Seconds $retryInterval
+                $lockExists = Get-AzResourceLock -LockName $lockName -Scope $lockScope -ErrorAction SilentlyContinue
+            } while ($lockExists)
```

Comment:
Wait'n'check for locks. As this removes the lock resource that was deployed, it logically is slightly different than the helper-function in this file.